task_sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -412,10 +414,6 @@ def _monitor_subprocess(self):
         - Sends heartbeats to the client to keep the task alive
         - Checks if the subprocess has exited
         """
-        # Until we have a selector for the process, don't poll for more than 10s, just in case it exists but
-        # doesn't produce any output
-        max_poll_interval = 10
```

Comment:
I just replaced this too with `MIN_HEARTBEAT_INTERVAL`. I do need to add `overtime` feature in the Supervisor -- will do in separate PR.

But let me know if you disagree with this @ashb

Reply:
This was previously to make sure that we call `self._process.wait(timeout=0)` at most every 10s (since we spend most of our time waiting on Readable events on the sockets.)

How ever that wasn't needed -- as the socket being closed, which happens when the subprocess exits even if it produces no output, shows up as a readable event (where upon reading gives the EOF) so that returns us from the select already.

Reply:
Makes sense, I added some more comments to that code for my understanding